src/ContactDetails.elm

```diff
@@ -0,0 +1,270 @@
+            [ Html.label
+                  [ Html.Attributes.for field.path
+                  , Html.Attributes.class "block text-sm font-medium leading-6 text-gray-900" ]
+                [ Html.text label ]
+            , Html.div [ Html.Attributes.class "mt-2"]
+                  [ Input.selectInput options field
+                      [ Html.Attributes.id field.path
+                      , Html.Attributes.class "block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:max-w-xs sm:text-sm sm:leading-6 disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none"
+                      , Html.Attributes.disabled disabled
+                      ]
+                ]
+            , errorFor field
+            ]
+
+
+textField : String -> String -> Bool -> Form () Address -> Html Form.Msg
+textField path label disabled form =
+    let
+        field =
+            Form.getFieldAsString path form
+    in
+        Html.div [ Html.Attributes.class "sm:col-span-full"]
+            [ Html.label
+                  [ Html.Attributes.for field.path
+                  , case field.liveError of
+                        Just _ ->
+                            Html.Attributes.class "block text-sm font-medium leading-6 text-red-700"
+
+                        Nothing ->
+                            Html.Attributes.class "block text-sm font-medium leading-6 text-gray-900"
+                  ]
+                [ Html.text label ]
+            , Html.div [ Html.Attributes.class "mt-2"]
+                [ Input.textInput field
+                      [ Html.Attributes.id field.path
+                      , Html.Attributes.class "block w-full rounded-md border-0 px-2 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6 disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none"
+                      , case field.liveError of
+                            Just _ ->
+                                Html.Attributes.class "bg-red-50 border border-red-500 text-red-900 placeholder-red-700 text-sm rounded-lg focus:ring-red-500 focus:border-red-500 block w-full p-2.5 dark:bg-red-100 dark:border-red-400"
+                          
+                            Nothing ->
+                                Html.Attributes.class ""
+                      , Html.Attributes.disabled disabled
+                      ]
+                ]
+            , errorFor field
+            ]
+
+
+errorFor : Form.FieldState () a -> Html msg
+errorFor field =
+    case field.liveError of
+        Just error ->
+            Html.div
+                [ Html.Attributes.class "mt-2 text-sm text-red-600" ]
+                [ Html.text (errorString error) ]
+                        
+        Nothing ->
+            Html.text ""
```

Comment:
NIT: I would use [`Html.Extra.viewMaybe`](https://package.elm-lang.org/packages/elm-community/html-extra/3.4.0/#viewMaybe).

Reply:
I general try to avoid adding unnecessary package dependencies beyond the core Elm packages if I can. In larger projects it can sometimes make sense, but in smaller project like this I think it is too much overhead with little benefit (and makes the code harder to read if you're not familiar with that particular helper library).

Reply:
That makes sense.